src/components/DemoBoardModal/DemoBoardModal.tsx

```diff
@@ -0,0 +1,434 @@
+        if (!isKomiOption(newKomiOption)) {
+            return;
+        }
+
+        const changedToCustom = newKomiOption === "custom" && komiOption !== "custom";
+        if (changedToCustom) {
+            setKomi(getDefaultKomi(ruleSet, false));
+        }
+        setKomiOption(newKomiOption);
+    }
+    function updateBoardSize(str: string) {
+        if (str === "custom") {
+            setIsCustomBoardSize(true);
+            return;
+        }
+
+        setIsCustomBoardSize(false);
+        const sizes = str.split("x");
+        setWidth(parseInt(sizes[0]));
+        setHeight(parseInt(sizes[1]));
+    }
+
+    const user = data.get("user");
+
+    function getValidatedSettings(): DemoSettings | null {
+        if (width === null || width < 1 || width > 25) {
+            document.getElementById("demo-board-modal-width")?.focus();
+            void alert.fire(_("Invalid board size, please correct and try again"));
+            return null;
+        }
+        if (height === null || height < 1 || height > 25) {
+            document.getElementById("demo-board-modal-height")?.focus();
+            void alert.fire(_("Invalid board size, please correct and try again"));
+            return null;
+        }
+        if (komiOption === "custom" && komi === null) {
+            document.getElementById("demo-board-modal-komi")?.focus();
+            void alert.fire(_("Please enter a number for komi."));
+            return null;
+        }
+
+        return {
+            name: name,
+            rules: ruleSet,
+            width: width,
+            height: height,
+            black_name: blackName,
+            black_ranking: blackRanking,
+            white_name: whiteName,
+            white_ranking: whiteRanking,
+            private: isPrivate,
+            komi_auto: komiOption,
+            ...(komi !== null && { komi: komi }),
+        };
+    }
+
+    async function createDemoBoard(): Promise<void> {
+        const validatedSettings = getValidatedSettings();
+        if (validatedSettings === null) {
+            return;
+        }
+
+        data.set("demo.settings", validatedSettings);
+
+        const black_pro = validatedSettings.black_ranking > 1000 ? 1 : 0;
+        const white_pro = validatedSettings.white_ranking > 1000 ? 1 : 0;
+        const extendedSettings = {
+            ...validatedSettings,
+            black_pro,
+            white_pro,
+            tournament_record_id: props.tournament_record_id,
+            tournament_record_round_id: props.tournament_record_round_id,
+            black_ranking: validatedSettings.black_ranking - (black_pro ? 1000 : 0),
+            white_ranking: validatedSettings.white_ranking - (white_pro ? 1000 : 0),
+            name: validatedSettings.name || _("Demo Board"),
+        };
+
+        await post("demos", extendedSettings)
+            .then((res) => {
+                console.log("Demo create response: ", res);
+                props.eventsRef.close();
```

Comment:
I've changed the functionality here a little bit. Currently the modal gets closed even if this request throws an error, but I think it's preferable to leave the modal open.